src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -1470,7 +1470,7 @@ private ImmutableArray<string> BuildSearchPaths(string sdkDirectoryOpt, List<str
 
         public static IEnumerable<string> ParseConditionalCompilationSymbols(string value, out IEnumerable<Diagnostic> diagnostics)
         {
-            Diagnostic myDiagnostic = null;
+            List<Diagnostic> myDiagnostics = new List<Diagnostic>();
```

Comment:
Or an ArrayBuilder? We usually don't use List of T.

---
In reply to: [100169189](https://github.com/dotnet/roslyn/pull/16920#discussion_r100169189) [](ancestors = 100169189)